app/models/operatingsystem.rb

```diff
@@ -190,27 +171,49 @@ def available_loaders
     ["None", "PXELinux BIOS"]
   end
 
-  # sets the prefix for the tfp files based on the os / arch combination
-  def pxe_prefix(arch)
-    "boot/#{self}-#{arch}".tr(" ", "-")
+  # Compatibility method, don't want to break all templates that use it.
+  def medium_uri(host)
+    Foreman::Deprecation.deprecation_warning("1.20", "medium_uri is now accessible through @medium_provider.medium_uri in templates")
+    @medium_provider = Foreman::Plugin.medium_providers.find_provider(host)
+    @medium_provider.medium_uri
+  end
```

Comment:
Please provide a template helper function if we don't have it already to avoid long `@host.bla.bla` and do patch into community templates, we have about 10 occurrences there, not many. It is worth changing it now.

Reply:
We already have one. Inside template you can call `medium_provider` and you will get it.